linkerd/proxy/tcp/src/forward.rs

```diff
@@ -27,23 +27,27 @@ impl<C> Forward<C> {
 impl<C, T, I> Service<(T, I)> for Forward<C>
 where
     C: Service<T>,
+    C::Response: Send + 'static,
+    C::Future: Send + 'static,
     C::Error: Into<Error>,
     C::Response: AsyncRead + AsyncWrite,
-    I: AsyncRead + AsyncWrite,
+    I: AsyncRead + AsyncWrite + Send + 'static,
 {
-    type Response = ();
+    type Response = Box<dyn Future<Item = (), Error = Error> + Send + 'static>;
     type Error = Error;
-    type Future = ForwardFuture<I, C::Future>;
+    type Future = Box<dyn Future<Item = Self::Response, Error = Error> + Send + 'static>;
```

Comment:
And this is always a `future::ok`, so again, it shouldn't need to be boxed
```suggestion
    type Future = futures::future::FutureResult<Self::Response, Error>;
```